test/spec_lint.rb

```diff
@@ -191,12 +191,12 @@ def obj.fatal(*) end
     lambda {
       Rack::Lint.new(valid_app).call(env("SERVER_NAME" => "A Host"))
     }.must_raise(Rack::Lint::LintError).
-      message.must_equal "A Host must be a valid authority"
+      message.must_equal "env[SERVER_NAME] must be a valid host: \"A Host\""
```

Comment:
The error message wording is kind of awkward, though I guess less awkward than before.  Maybe something like (I think we should also quote the key):

```suggestion
      message.must_equal "env['SERVER_NAME'] is an invalid host: \"A Host\""
```

Similar for HTTP_HOST

Reply:
> `SERVER_NAME` is a constant. I feel like the quotes just add extra noise and don't really add anything of value. It's also consistent with other errors. If you want to change that, I suggest we merge this as is and then update it in a separate PR.

`SERVER_NAME` isn't a valid constant.  `Rack::SERVER_NAME` is, but that isn't what is listed.  However, if this is consistent with other errors, I agree there is no need to fix it in this PR. I would still be in favor of fixing it later.

> Regarding the actual error, I think the `LintError` should specifically call out the requirement, as in, the server name must be a valid host. If you say "Server name is invalid host", then it begs the question "why does that matter?". When we say "Server name must be a valid host" we are clearly describing the requirement.

The issue with `must be a valid host` is that is awkward as it ends with the provided, invalid host, with no statement that it is invalid.  Users can probably infer it, but I don't think it is obvious.  I'm fine keeping `must be a valid host`, as long as additional wording is added.  Maybe 

```suggestion
      message.must_equal "env['SERVER_NAME'] must be a valid host. Invalid host provided: \"A Host\""
```

Reply:
I am okay with:

```
env[SERVER_NAME] must be a valid host, given \"A Host\"
```

I think there is value keeping the error message terse.

Reply:
> I think there is value keeping the error message terse.

Can you explain why terseness in this area is beneficial?  Given the choice between clarity and terseness, I would choose clarity for Lint's exception messages.  Even if users can probably infer correctly, I don't think it is beneficial to force the need for such inference on them, when we could instead make the error message more clear.

Reply:
I think most people prefer terse error messages, myself included. I think this discussion and change is getting outside of the scope of this PR. This PR originally proposed error messages that are consistent with adjacent and similar error messages. Why don't we go with what was originally proposed (and consistent with others) and then if you feel strongly to reword all the error messages, open a new PR. I'm fine with making the messages clearer.

Reply:
Looking at the adjacent error messages, I think it would be consistent if you dropped the display of the invalid data.  So just: `env[SERVER_NAME] must be a valid authority` (and similar for HTTP_HOST).  We don't show invalid protocols or ports, so I guess there is no reason we must show invalid hosts or authorities.

Reply:
Okay, I will do that.